src/shared_memory_manager.cc

```diff
@@ -264,6 +264,45 @@ OpenCudaIPCRegion(
 
   return nullptr;
 }
+
+TRITONSERVER_Error*
+GetCudaSharedMemoryRegionSize(CUdeviceptr data_ptr, size_t& shm_region_size)
+{
+  CUdeviceptr* base = nullptr;
+  CUresult result = cuMemGetAddressRange(base, &shm_region_size, data_ptr);
+  if (result != CUDA_SUCCESS) {
+    const char* errorString;
+    if (cuGetErrorString(result, &errorString) != CUDA_SUCCESS) {
+      return TRITONSERVER_ErrorNew(
+          TRITONSERVER_ERROR_INTERNAL, "Failed to get CUDA error string");
```

Comment:
```suggestion
          TRITONSERVER_ERROR_INTERNAL, "Failed to get shared memory information");
```

Reply:
same here - let's log details - but error be generic

Reply:
This should still involve CUDA because user could request both system shm and CUDA shm right?

Reply:
Please see here for the context of removing `CUDA`: https://github.com/triton-inference-server/server/pull/7178#discussion_r1591672353